src/proxy/mod.rs

```diff
@@ -27,11 +28,12 @@ impl Proxy {
         cfg: config::Config,
         workloads: WorkloadInformation,
         secret_manager: identity::SecretManager,
+        drain: Watch,
     ) -> Result<Proxy, Error> {
         // We setup all the listeners first so we can capture any errors that should block startup
         let inbound_passthrough = InboundPassthrough::new(cfg.clone());
         let inbound = Inbound::new(cfg.clone(), workloads.clone(), secret_manager.clone()).await?;
-        let outbound = Outbound::new(cfg.clone(), secret_manager, workloads).await?;
+        let outbound = Outbound::new(cfg.clone(), secret_manager, workloads, drain).await?;
```

Comment:
at some point as we keep passing the `drain` down, does it make sense to clone it (even though there aren't other uses)

makes it clear that each consumer should take a clone. i think a codebase for a language we're not that familiar with it's nice when you're surrounded by examples of how you can get the same functionality. 

random thought, not sure if it actually makes much sense. the clone should be nearly free.

Reply:
Yeah, probably so